include/roboteam_proto/Subscriber.h

```diff
@@ -94,8 +88,21 @@ class Subscriber {
     t1 = std::thread(&Subscriber::poll, this);
   }
 
-  ~Subscriber();
-  void poll();
+  ~Subscriber() {
+    std::cout << "[Roboteam_proto] Stopping subscriber for " << channel.name << std::endl;
+    running = false;
+    t1.join();
+    reactor->remove(*socket);
+    socket->close();
+    delete socket;
+    delete reactor;
+  }
+
+  void poll() {
+    while (running) {
+      reactor->poll(500);
```

Comment:
What is the input argument 500 doing here? I'm guessing microseconds, but a comment would be nice.

Reply:
it's milliseconds. And admittedly it's quite a random number < 1 second ;-) 

I have to constantly keep polling, but first this parameter was blocking, so if I would destruct a subscriber I could not join this thread because `reactor->poll()` would block. This parameter gives it a timeout so every 500ms it breaks out and polls again. Behavior is the same but now i can destruct the object properly at least.

Maybe I can decrease it to 100, this counter usually resets when there is a new message anyway.

Reply:
The comment makes it much clearer, thanks.